lib/ignored-paths.js

```diff
@@ -92,14 +92,24 @@ class IgnoredPaths {
         }
 
         /**
+         * read ignore filepath
+         * @param {string} filepath, file to add to ig
+         * @returns {array} raw ignore rules
+         */
+        function readIgnoreFile(filepath) {
+            return fs.readFileSync(filepath, "utf8");
+        }
```

Comment:
I think it would be better to avoid doing that. Conceptually, it seems like it would be simpler for the cache to be an implementation detail of `readIgnoreFile`. That way, everything else can just treat `readIgnoreFile` as an abstraction for "get the contents of this file somehow", without worrying about the details of how/whether files are cached.

This might be easier if you make `addIgnoreFile` and `readIgnoreFile` class methods in `IgnoredPaths` (and change calls to use `this.addIgnoreFile()` rather than `addIgnoreFile()`), because then `readIgnoreFile` will be able to deal with `this.cache` on its own rather than accepting it as an argument.